rules/CodingStyle/Application/UseImportsAdder.php

```diff
@@ -201,35 +228,46 @@ private function diffFullyQualifiedObjectTypes(array $mainTypes, array $typesToR
 
     /**
      * @param array<AliasedObjectType|FullyQualifiedObjectType> $useImportTypes
-     * @param array<FullyQualifiedObjectType|AliasedObjectType> $constantUseImportTypes
      * @param array<FullyQualifiedObjectType|AliasedObjectType> $functionUseImportTypes
+     * @param array<FullyQualifiedObjectType|AliasedObjectType> $constantUseImportTypes
      * @return Use_[]
      */
     private function createUses(
         array $useImportTypes,
-        array $constantUseImportTypes,
         array $functionUseImportTypes,
+        array $constantUseImportTypes,
         ?string $namespaceName
     ): array {
         $newUses = [];
 
         /** @var array<Use_::TYPE_*, array<AliasedObjectType|FullyQualifiedObjectType>> $importsMapping */
         $importsMapping = [
             Use_::TYPE_NORMAL => $useImportTypes,
-            Use_::TYPE_CONSTANT => $constantUseImportTypes,
             Use_::TYPE_FUNCTION => $functionUseImportTypes,
+            Use_::TYPE_CONSTANT => $constantUseImportTypes,
         ];
 
         foreach ($importsMapping as $type => $importTypes) {
-            /** @var AliasedObjectType|FullyQualifiedObjectType $importType */
+            $newUsesPerType = [];
             foreach ($importTypes as $importType) {
                 if ($namespaceName !== null && $this->isCurrentNamespace($namespaceName, $importType)) {
                     continue;
                 }
 
                 // already imported in previous cycle
-                $newUses[] = $importType->getUseNode($type);
+                $newUsesPerType[] = $importType->getUseNode($type);
+            }
+
+            if (SimpleParameterProvider::provideBoolParameter(Option::IMPORT_INSERT_SORTED)) {
```

Comment:
If the option to insert the use statements in sorted order is set, we sort them alphabetically before adding them to the list of `Use_` statements to add. We do this in groups of types. This process guarantees that the return of this function returns these statements in sorted order, and with the right order for the use types